### PR TITLE
fix: Error pages can be translated

### LINKED ISSUE #2409: 🐛 [Bug] User-not-found-page not translated
> [datenbrei](https://github.com/datenbrei) **Authored by [datenbrei](https://github.com/datenbrei)**
_<time datetime="2019-12-03T08:36:55Z" title="Tuesday, December 3rd 2019, 9:36:55 am +01:00">Dec 3, 2019</time>_
_Closed <time datetime="2020-02-07T10:59:34Z" title="Friday, February 7th 2020, 11:59:34 am +01:00">Feb 7, 2020</time>_
---

## :bug: Bugreport
The page "User not found" is not yet translated.

user

### Steps to reproduce the behavior
1. Call a user profile which had been deleted

### Expected behavior
The Page should
1. Be translated
2. Explain, why this page comes up. For example: "The user deleted his own account".

### Version & Environment
any

### PR DESCRIPTION
**Pull request by Tirokk**
> [Mogge](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-01-17T17:33:40Z" title="Friday, January 17th 2020, 6:33:40 pm +01:00">Jan 17, 2020</time>_
_Merged <time datetime="2020-02-07T10:59:34Z" title="Friday, February 7th 2020, 11:59:34 am +01:00">Feb 7, 2020</time>_
---

## 🍰 Pullrequest
grafik

### Issues
- fixes #2409

### CODE REVIEW

**Comment by Tirokk** [comment]
> [Mogge](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-01-17T17:36:37Z" title="Friday, January 17th 2020, 6:36:37 pm +01:00">Jan 17, 2020</time>_

---


@alina-beck do you have suggestions for a better layout. May be a nice image ...

**Comment by Tirokk** [comment]
> [cypress[bot]](https://github.com/apps/cypress) **Authored by [cypress[bot]](https://github.com/apps/cypress)**
_<time datetime="2020-01-17T17:58:54Z" title="Friday, January 17th 2020, 6:58:54 pm +01:00">Jan 17, 2020</time>_

---



<p align="center">
  <br />
  
</p>

---

### Test summary
<a href="https://dashboard.cypress.io/projects/qa7fe2/runs/4622/specs"></a> <b title="Passed">53</b>  &bull; <a href="https://dashboard.cypress.io/projects/qa7fe2/runs/4622/failures"></a> <b title="Failed">0</b>  &bull;  <b title="Pending">0</b> &bull;  <b title="Skipped">0</b>

---

### Run details
| | |
| --- | --- |
| **Project** | Human-Connection |
| **Status** | Passed |
| **Commit** | c86132de21 |
| **Started** | Jan 29, 2020 10:32 AM |
| **Ended** | Jan 29, 2020 10:45 AM |
| **Duration** | 12:29 <a href="https://on.cypress.io/parallelization" title="Enable parallelization to reduce run duration">💡</a> |
| **OS** | Linux Ubuntu Linux - 16.04 |
| **Browser** | Electron 78  |

[View run in Cypress Dashboard ➡️](https://dashboard.cypress.io/projects/qa7fe2/runs/4622/specs)



---

This comment has been generated by **cypress-bot** as a result of this project's GitHub integration settings. You can manage this integration in [this project's settings in the Cypress Dashboard](https://dashboard.cypress.io/projects/qa7fe2/settings)

**Comment by Tirokk** [comment]
> [alina-beck](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2020-01-20T09:47:45Z" title="Monday, January 20th 2020, 10:47:45 am +01:00">Jan 20, 2020</time>_

---


yeah, an image would be nice – you could check the `/static/img/` folder to see if you can find one that fits?

**Comment by Tirokk** [comment]
> [Mogge](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-01-20T10:32:04Z" title="Monday, January 20th 2020, 11:32:04 am +01:00">Jan 20, 2020</time>_

---


Now it looks like this:
image

I took the `exclamation-circle` icon.
That do you think @alina-beck ?

**Comment by Tirokk** [comment]
> [Mogge](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-01-20T12:15:45Z" title="Monday, January 20th 2020, 1:15:45 pm +01:00">Jan 20, 2020</time>_

---


One question. There are some 403 from webapp middleware. Should they be localised too?
There seems to be a generated 404 which so far I could not find in the Codebase. It appears when there is an URL which is not mapped at all. Where could I find this message?

File: webapp/layouts/error.vue [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<template>
+  <div class="error-container">
+    <img class="error-image" :src="image" />
+    <br />
+    <span class="error-message">{{ $t(message) }}</span>
```
Comment (Tirokk) [inline webapp/layouts/error.vue:5]: > [Mogge](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-01-24T13:51:28Z" title="Friday, January 24th 2020, 2:51:28 pm +01:00">Jan 24, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,34 @@
+<template>
+  <div class="error-container">
+    <base-icon class="error-icon" name="exclamation-circle" />
+    <br />
+    <span v-if="error.message === 'This page could not be found'" class="error-message">
```


I was not able to locate the place where the default 404 error message comes from. So I catch it here. Any ideas?

Comment (Tirokk) [reply webapp/layouts/error.vue:5]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-01-24T14:13:23Z" title="Friday, January 24th 2020, 3:13:23 pm +01:00">Jan 24, 2020</time>_

---


You could dispatch the error code:

```js
computed: {
  message() {
    const mapping = {
       403: 'error-pages.not-authorized',
       404: // ...
       500: // ...
    }
    const key = mapping(this.error.code) || 'error.default'
    return this.$(key)
  }
}

File: webapp/middleware/isAdmin.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 export default ({ store, error }) => {
   if (!store.getters['auth/isAdmin']) {
-    return error({ statusCode: 403 })
+    return error({ statusCode: 403, message: 'error-pages.not-authorized' })
```
Comment (Tirokk) [inline webapp/middleware/isAdmin.js:3]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-01-24T14:15:35Z" title="Friday, January 24th 2020, 3:15:35 pm +01:00">Jan 24, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,5 +1,5 @@
 export default ({ store, error }) => {
   if (!store.getters['auth/isAdmin']) {
-    return error({ statusCode: 403 })
+    return error({ statusCode: 403, message: 'error-pages.not-authorized' })
```


I'm having difficulties with `message` here. It's not the message, it's the translation key.

So, on the consuming side, I would check if there is a translation key for `error.message` (check if it's a key or if the error message is coming from NuxtJS). Just to make sure that we're not trying to translate a default Nuxt error message here.

File: webapp/layouts/error.vue [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        500: 'error-pages.500-default',
+        503: 'error-pages.503-default',
+      }
+      return this.error.key || mapping[this.error.statusCode] || 'error-pages.default'
+    },
+    image() {
+      return `/img/svg/errors/error${this.error.statusCode || '500'}.svg`
+    },
+  },
+}
```
Comment (Tirokk) [inline webapp/layouts/error.vue:28]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-01-28T21:06:57Z" title="Tuesday, January 28th 2020, 10:06:57 pm +01:00">Jan 28, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,50 @@
+<template>
+  <div class="error-container">
+    <img class="error-image" :src="image" />
+    <br />
+    <span v-if="error.message === 'This page could not be found'" class="error-message">
+      {{ $t('error-pages.default') }}
+    </span>
+    <span v-else class="error-message">{{ $t(error.message) }}</span>
+    <br />
+    <nuxt-link to="/">{{ $t('error-pages.back-to-index') }}</nuxt-link>
+  </div>
+</template>
+
+<script>
+export default {
+  props: ['error'],
+  computed: {
+    message() {
+      const mapping = {
+        403: 'error-pages.403-default',
+        404: 'error-pages.404-default',
+        500: 'error-pages.500-default',
+        503: 'error-pages.503-default',
+      }
+      return this.error.key || mapping[this.error.statusCode] || 'error-pages.default'
+    },
+    image() {
+      return `/img/svg/errors/error${this.error.statusCode ? this.error.statusCode : '500'}.svg`
```


```suggestion
      return `/img/svg/errors/error${this.error.statusCode || '500'}.svg`
```

File: webapp/locales/de.json [hunk 4]
```diff
@@ -798,5 +798,16 @@
         "donations-for": "Spenden für",
         "donate-now": "Jetzt spenden",
         "amount-of-total": "{amount} von {total} € erreicht"
+    },
+    "error-pages" : {
+	"profile-not-found": "Dieses Profil konnte nicht gefunden werden",
+	"back-to-index": "Zurück zur Startseite",
+	"post-not-found": "Dieser Beitrag konnte nicht gefunden werden",
+	"cannot-edit-post": "Dieser Beitrag kann nicht editiert werden",
```
Comment (Tirokk) [inline webapp/locales/de.json:806]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-01-28T21:07:08Z" title="Tuesday, January 28th 2020, 10:07:08 pm +01:00">Jan 28, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -798,5 +798,16 @@
         "donations-for": "Spenden für",
         "donate-now": "Jetzt spenden",
         "amount-of-total": "{amount} von {total} € erreicht"
+    },
+    "error-pages" : {
+	"profile-not-found": "Dieses Profil konnte nicht gefunden werden",
+	"back-to-index": "Zurück zur Startseite",
+	"post-not-found": "Dieser Beitrag konnte nicht gefunden werden",
+        "cannot-edit-post": "Dieser Beitrag kann nicht editiert werden",
```


Fix indentation

File: webapp/layouts/error.vue (hunk 3 above)
Comment (Tirokk) [reply webapp/layouts/error.vue:28]: > [Mogge](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-01-28T21:28:21Z" title="Tuesday, January 28th 2020, 10:28:21 pm +01:00">Jan 28, 2020</time>_

---


@roschaefer I thought you wanted an image :)

**Review by Mogge — CHANGES_REQUESTED** [review]
Screenshot - 2020-01-28T221535 452
Screenshot - 2020-01-28T221518 658


Maybe a little large?

blown away

**Review by Mogge — APPROVED** [review]
Alright I haven't looked how the error pages look now but I can see from the code that everything is gonna be OK.

everything is gonna be OK

**Pull request closed without merging**